src/core/time.d

```diff
@@ -1927,35 +1927,68 @@ unittest
 
 unittest
 {
-    auto t1v = 1000;
-    auto t2v = 333;
-    auto t1 = TickDuration.from!"seconds"(t1v);
-    auto t2 = TickDuration.from!"seconds"(t2v);
+    void testFun(string U)() {
+        auto t1v = 1000;
+        auto t2v = 333;
 
-    foreach(F; _TypeTuple!(int,uint,long,ulong,float,double,real))
-    {
-        F t1f = to!("seconds",F)(t1);
-        F t2f = to!("seconds",F)(t2);
-        auto t12d = t1 / t2v;
-        auto t12m = t1 - t2;
-        F t3f = to!("seconds",F)(t12d);
-        F t4f = to!("seconds",F)(t12m);
+        auto t1 = TickDuration.from!U(t1v);
+        auto t2 = TickDuration.from!U(t2v);
 
-        static if(is(F : float))
-        {
-            assert((t1f - cast(F)t1v) < 0.0001);
-            assert((t2f - cast(F)t2v) < 0.0001);
-            assert(t3f - (cast(F)t1v) / (cast(F)t2v) < 0.0001);
-            assert(t4f - (cast(F)(t1v - t2v)) < 0.0001);
-        }
-        else
+        foreach (F; _TypeTuple!(int,uint,long,ulong,float,double,real))
         {
```

Comment:
It's usually better to turn those foreach into a function literal to mitigate the expensive O(N^2) optimizations of the backend.